src/jobs/models/jobManager.ts

```diff
@@ -46,8 +48,84 @@ export class JobManager {
     }
   }
 
-  private convertPrismaToJobResponse(prismaObjects: Prisma.JobGetPayload<Record<string, never>>): IJobModel {
-    const jobObject: IJobModel = {
+  public async getJobById(jobId: string): Promise<JobModel> {
+    const queryBody = {
+      where: {
+        id: jobId,
+      },
+    };
+
+    const job = await this.prisma.job.findUnique(queryBody);
+
+    if (!job) {
+      throw new JobNotFoundError('JOB_NOT_FOUND');
+    }
+
+    return this.convertPrismaToJobResponse(job);
+  }
+
+  public async updateUserMetadata(jobId: string, userMetadata: Record<string, unknown>): Promise<void> {
+    const updateQueryBody = {
+      where: {
+        id: jobId,
+      },
+      data: {
+        userMetadata: userMetadata,
+      },
+    };
+
+    try {
+      await this.prisma.job.update(updateQueryBody);
+    } catch (err) {
+      if (err instanceof Prisma.PrismaClientKnownRequestError && err.code === 'P2025') {
+        throw new JobNotFoundError('JOB_NOT_FOUND');
+      }
+      throw err;
+    }
+  }
+
+  public async updatePriority(jobId: string, priority: Priority): Promise<void> {
+    const updateQueryBody = {
+      where: {
+        id: jobId,
+      },
+      data: {
+        priority,
+      },
+    };
+
+    try {
+      await this.prisma.job.update(updateQueryBody);
+    } catch (err) {
+      if (err instanceof Prisma.PrismaClientKnownRequestError && err.code === 'P2025') {
+        throw new JobNotFoundError('JOB_NOT_FOUND');
+      }
+      throw err;
+    }
+  }
+
+  public async updateStatus(jobId: string, status: OperationStatus): Promise<void> {
+    const updateQueryBody = {
+      where: {
+        id: jobId,
+      },
+      data: {
+        status,
+      },
+    };
+
+    try {
+      await this.prisma.job.update(updateQueryBody);
+    } catch (err) {
+      if (err instanceof Prisma.PrismaClientKnownRequestError && err.code === 'P2025') {
```

Comment:
@CptSchnitz  https://www.npmjs.com/package/prisma-error-enum
should I use something like that? instead of hardcoded code?